src/hotspot/share/gc/g1/g1ConcurrentMark.hpp

```diff
@@ -136,10 +136,69 @@ class G1CMMarkStack {
     G1TaskQueueEntry data[EntriesPerChunk];
   };
 
-  size_t _max_chunk_capacity;    // Maximum number of TaskQueueEntryChunk elements on the stack.
+  class ChunkAllocator {
+    size_t _min_capacity;
+    size_t _max_capacity;
+    size_t _capacity;
+    size_t _num_buckets;
+    bool _growable;
```

Comment:
I wonder if this can be named `_should_grow`.

Reply:
Fixed